pkg/controller/worker/machines.go

```diff
@@ -114,23 +116,33 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			Image:   machineImage,
 		})
 
-		volumeSize, err := worker.DiskSize(pool.Volume.Size)
+		disks := make([]map[string]interface{}, 0)
+		// root volume
+		disk, err := createDiskSpec(*pool.Volume, w.worker.Name, machineImage, true)
 		if err != nil {
 			return err
 		}
 
-		disk := map[string]interface{}{
-			"autoDelete": true,
-			"boot":       true,
-			"sizeGb":     volumeSize,
-			"type":       pool.Volume.Type,
-			"image":      machineImage,
-			"labels": map[string]interface{}{
-				"name": w.worker.Name,
-			},
+		disks = append(disks, disk)
```

Comment:
I know it wasn't like this before, but can we add a `nil` pointer check to prevent panics if `pool.Volume` is not set?

Reply:
Accept